File: perl-programming/examples/regex/parse_field_value_pairs.pl
#!/usr/bin/perl
use strict;
use warnings;

# data: field_value_pairs.txt
my $filename = shift or die "Usage: $0 filename\n";

open(my $fh, "<", $filename) or die "Could not open '$filename'\n";
while (my $line = <$fh>) {
    chomp $line;
    my ($field, $value) = split /\s*=\s*/, $line;
    print "$value=$field\n";
}

File: python-regular-expressions/src/examples/regex/parse_field_value_pairs.py
import sys
import re

# data: field_value_pairs.txt
if len(sys.argv) != 2:
    exit(f"Usage: {sys.argv[0]} filename")

filename = sys.argv[1]

with open(filename) as fh:
    for line in fh:
        line = line.rstrip("\n")
        field, value = re.split(r'\s*=\s*', line)
        print(f"{value}={field}")
